### PR TITLE
ENH: add geopandas.list_layers wrapping pyogrio

### LINKED ISSUE #2837: ENH: Function to show layers in a particular file
As far as I can tell, the only way to see the layers in a particular file is to read the file first, e.g.

```python
import geopandas as gpd

# read the GeoPackage file
gdf = gpd.read_file('your_geopackage_file.gpkg')

# list the layer names
layer_names = list(gdf.name)
print(layer_names)
```

Would it be possible/useful to add a function that takes the file name as the first argument and returns the layers? This from the `sf` package could provide some inspiration: https://r-spatial.github.io/sf/reference/st_layers.html

Apologies in the like event I've missed something, fairly new to this space!

### PR DESCRIPTION
**Pull request by martinfleis**
Closes #2837

Marking it as a draft for now as before I'll start touching the docs, I'd like to get #3223 in to avoid unnecessary conflicts.

The only thing this does is to wrap `pyogrio.list_layers` and return it as a DataFrame instead of an array to avoid confusion like the one in https://github.com/geopandas/pyogrio/issues/275/. I was also considering returning a Series indexed by a layer name but DataFrame _feels_ more right in this case.

To-do:
- [x] documentation

### CODE REVIEW

File: geopandas/io/file.py [hunk 1]
```diff
@@ -796,3 +796,34 @@ def _geometry_types(df):
         geom_types = geom_types[0]
 
     return geom_types
+
+
+def _list_layers(filename) -> pd.DataFrame:
+    """List layers available in a file.
+
+    Provides an overview of layers available in a file or URL toghether with their
```
Comment (brendan-ward) [inline geopandas/io/file.py:804]:
```suggestion
    Provides an overview of layers available in a file or URL together with their
```

File: geopandas/io/file.py [hunk 2]
```diff
@@ -796,3 +796,34 @@ def _geometry_types(df):
...
+    geometry types. When supported, this includes both spatial and non-spatial layers.
+    Non-spatial layers are indicated by the ``"geometry_type"`` column being ``None``.
+    GeoPandas will not read such layers but they can be read into pd.DataFrame using
```
Comment (brendan-ward) [inline geopandas/io/file.py:807]:
```suggestion
    GeoPandas will not read such layers but they can be read into a pd.DataFrame using
```

File: geopandas/io/file.py [hunk 3]
```diff
@@ -796,3 +796,34 @@ def _geometry_types(df):
         geom_types = geom_types[0]
 
     return geom_types
+
+
+def _list_layers(filename) -> pd.DataFrame:
+    """List layers available in a file.
+
+    Provides an overview of layers available in a file or URL toghether with their
+    geometry types. When supported, this includes both spatial and non-spatial layers.
```
Comment (brendan-ward) [inline geopandas/io/file.py:805]:
```suggestion
    geometry types. When supported by the data source, this includes both spatial and non-spatial layers.
```

File: geopandas/io/file.py [hunk 4]
```diff
@@ -796,3 +796,34 @@ def _geometry_types(df):
...
+    filename : str, path object or file-like object
+        Either the absolute or relative path to the file or URL to
+        be opened, or any object with a read() method (such as an open file
+        or StringIO)
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame with columns "name" and "geometry_type" and one row per layer.
+    """
```
Comment (brendan-ward) [inline geopandas/io/file.py:821]: Is it worth adding an example of how to get just the layer names as a list?

E.g., 
`geopandas.list_layers(<path>).name.tolist()`

The DataFrame return seems like the cleanest way to handle the pairs of layer name and geometry type, but I wonder if folks will sometimes just want the layer names.

**Review by brendan-ward — COMMENTED** [review]
Thanks for working on this!

**Review by theroggy — APPROVED** [review]

**Review by m-richards — APPROVED** [review]
Thanks Martin, looks good to me

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**